pkg/pipelines/webhook/webhook.go

```diff
@@ -97,17 +104,50 @@ func newWebhookInfo(accessToken, pipelinesFile string, serviceName *QualifiedSer
 		return nil, err
 	}
 
+	listenerURL, err := getListenerURL(clusterResources, cicdNamepace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get event listener URL: %v", err)
+	}
+	if accessToken == "" {
+		accessToken, err = getAccessToken(gitRepoURL)
+		if err != nil {
+			return nil, err
+		}
+	}
 	repository, err := git.NewRepository(gitRepoURL, accessToken)
 	if err != nil {
 		return nil, err
 	}
+	return &webhookInfo{clusterResources, repository, gitRepoURL, cicdNamepace, listenerURL, accessToken, serviceName, isCICD}, nil
+}
 
-	listenerURL, err := getListenerURL(clusterResources, cicdNamepace)
+func getAccessToken(gitRepoURL string) (string, error) {
+	hostName, err := HostFromURL(gitRepoURL)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get event listener URL: %v", err)
+		return "", err
 	}
-
-	return &webhookInfo{clusterResources, repository, gitRepoURL, cicdNamepace, listenerURL, accessToken, serviceName, isCICD}, nil
+	accessToken, err := keyring.Get(KeyringServiceName, hostName)
+	if err != nil && err != keyring.ErrNotFound {
+		return "", err
+	}
+	if err != nil && err == keyring.ErrNotFound {
+		parsed, err := url.Parse(gitRepoURL)
```

Comment:
I thought the mapping is hostname -> token in your env var and your already have the hostName above.